servers/visual/visual_server_scene.cpp

```diff
@@ -1400,7 +1400,16 @@ bool VisualServerScene::_light_instance_update_shadow(Instance *p_instance, cons
 
 					float w, h;
 					p_cam_projection.get_viewport_size(w, h);
-					camera_matrix.set_orthogonal(w, aspect, distances[(i == 0 || !overlap) ? i : i - 1], distances[i + 1], false);
+
+					real_t size;
+					if (w > 0.0) {
+						size = w * ((real_t)h / w) * 2.0;
```

Comment:
Is the cast to `real_t` necessary when they're both already `float`?

Reply:
Barring any precision errors this is just `2.0 * h`, no?

Reply:
It probably doesn't matter one iota, I'm just trying to get into the whole 'everything should be a real_t to future proof for double precision' idea. set_orthogonal takes a real_t as argument, and h / w should be minutely more accurate as a double lol. I'll consider using floats when I edit tomorrow. :smile:

Reply:
> Barring any precision errors this is just `2.0 * h`, no?

Good point, I didn't even notice that. Will address tomorrow. :)